### DISCUSSION TITLE
Can you query the current InvenTree version from a Plugin?

### DISCUSSION THREAD

**Discussion by LavissaWoW** [inventree/InvenTree#6104] [Q&A] [answered]
As the question says.
Is it possible for a plugin to check the current InvenTree version?
This would be highly useful for plugins having functionality that can, may, or have changed over the course of releases.

**Comment by SchrodingersGat** [comment, marked as answer]
```python
from InvenTree.version import inventreeVersion, inventreeApiVersion

sw_version = inventreeVersion()
api_version = inventreeApiVersion()
```

**Reply by LavissaWoW** [reply]
Thanks! 
Figured I'd add some version check if I'm uploading the small label context plugin.

**Reply by matmair** [reply]
If you just need to pin the version of InvenTree the plugin runs with you can set 'MIN_VERSION' and 'MAX_VERSION' in thee plugins meta section - as listed [here](https://docs.inventree.org/en/stable/extend/plugins/?h=max_version#plugin-options). Works without any custom code.
If you need more intricate checks make sure to put them into the __init__ function of the plugin class and raise an error there. That way the plugin never gets loaded / activated and you can be sure it will not run.

**Reply by LavissaWoW** [reply]
> If you just need to pin the version of InvenTree the plugin runs with you can set 'MIN_VERSION' and 'MAX_VERSION' in thee plugins meta section - as listed [here](https://docs.inventree.org/en/stable/extend/plugins/?h=max_version#plugin-options). Works without any custom code.
> 
> If you need more intricate checks make sure to put them into the __init__ function of the plugin class and raise an error there. That way the plugin never gets loaded / activated and you can be sure it will not run.

Ooh, appreciate the tip! 🤩 The plan is to include the functionality of this small plugin in core. So I don't want it to interfere after its functionality has been added to core.

**Comment by LavissaWoW** [comment]
@SchrodingersGat Speaking of versions, has there been any considerations made in terms of preventing the Django backend from starting if you have pending migrations, outdated packages etc? Possibly a command line warning etc?

Like, you really should run `invoke update`

I spoke with a colleague who uses InvenTree, and he'd made the mistake of pulling down new commits, forgetting he had done so, and having a hard time working out why InvenTree wouldn't run properly.

**Reply by SchrodingersGat** [reply]
@LavissaWoW an interesting idea. We have the `auto_update` configuration option (default `False`) which will auto run migrations on start. We could check if migrations are present and if this is not set.

We also display a `pending migrations` warning in the UI 

I'd be happy to put a warning message into the command line also… but doesn't Django already so this?

**Reply by LavissaWoW** [reply]
I'll construct a few environments to see what warnings you get both on the CLI side and the GUI side, and I'll update you. I found the concept interesting too.
